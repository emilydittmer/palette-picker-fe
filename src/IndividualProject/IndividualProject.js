import React, { Component } from 'react';
import './IndividualProject.scss'
import { fetchPalettesInProject } from '../utils/apiCalls'
import PaletteInProject from '../PaletteInProject/PaletteInProject'

class IndividualProject extends Component {
  constructor(project){
    super(project);
    this.state={
      palettes:[]
    }
  }

  async componentDidMount(){
    const palettes = await fetchPalettesInProject(this.props.id)
    this.setState({ palettes })
  }

  render(){
    const allPalettes = this.state.palettes.map(palette => {
      return (
        <PaletteInProject 
          key={palette.id}
          id={palette.id}
          projectId={palette.project_id}
          color1={palette.color_1}
          color2={palette.color_2}
          color3={palette.color_3}
          color4={palette.color_4}
          color5={palette.color_5}
        />
      )
    })
    return(
      <article>
        <h1>{this.props.title}</h1>
          {allPalettes}
      </article>
    )
  }
}

export default IndividualProject;