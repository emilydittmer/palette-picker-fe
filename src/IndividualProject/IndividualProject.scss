.all-palettes{
  display: flex;
  margin: 10px;
}

.individual-project {
  overflow-x: scroll;
}