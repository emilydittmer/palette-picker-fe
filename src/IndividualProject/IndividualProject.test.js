import React from 'react'
import { shallow } from 'enzyme'
import { IndividualProject, mapDispatchToProps, mapStateToProps } from './IndividualProject'

describe('IndividualProject', () => {
  let wrapper, props;
  beforeEach(() => {
    props = {
      id: 1,
      deletePalette: jest.fn(() => ([{id: 1}])),
    }
    wrapper = shallow(<IndividualProject {...props} />)
  })
  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('Should have state', () => {
    const expected = {
      palettes: [],
      error: ""
    }
    expect(wrapper.state()).toEqual(expected)
  })
  it('HAPPY PATH: Should call a deletePalette function', async () => {
    await wrapper.instance().deletePalette(1)
    const expected = [{id: 1}]
    expect(wrapper.state().palettes).toEqual(expected)
  })
  it('SAD PATH: Should call a deletePalette function', async () => {
    const props = {
      id: 1,
      deletePalette: jest.fn(() => undefined)
    }
    wrapper = shallow(<IndividualProject {...props}/>)
    await wrapper.instance().deletePalette()
    const expected = 'Error: No palettes in project'
    expect(wrapper.state().error).toEqual(expected)
  })
})

describe('mapStateToProps', () => {
  it('Should return the store', () => { 
    const expected = {id: 1}
    const result = mapStateToProps({id: 1})

    expect(result).toEqual(expected)
  })
})

describe('mapDispatchToProps', () => {
  
})