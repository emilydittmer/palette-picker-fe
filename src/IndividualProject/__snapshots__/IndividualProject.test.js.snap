// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndividualProject Should match the snapshot 1`] = `ShallowWrapper {}`;
